packages/python/plotly/plotly/basedatatypes.py

```diff
@@ -1559,17 +1559,26 @@ def _add_annotation_like(
                         subplot_type=refs[0].subplot_type,
                     )
                 )
-            if len(refs) == 1 and secondary_y:
-                raise ValueError(
-                    """
-Cannot add {prop_singular} to secondary y-axis of subplot at position ({r}, {c})
-because subplot does not have a secondary y-axis"""
-                )
-            if secondary_y:
-                xaxis, yaxis = refs[1].layout_keys
+
+            # If the new_object was created with an yref specified, the specified yref should be used otherwise assign the xref and yref from the layout_keys
```

Comment:
```suggestion
            # If new_obj was created with yref specified, the specified yref
            # should be used, otherwise assign xref and yref from layout_keys
```